fix scaling of WEffectSelector and WBeatSpinBox

Be-ing opened the pull request:
~~This was easier than expected. :)~~ This was pretty tricky to find the bug.

Before:
![image](https://user-images.githubusercontent.com/9455094/36543586-c47e1f48-17a9-11e8-8130-7ffbf3181c95.png)

After:
![screenshot from 2018-02-22 08-14-21](https://user-images.githubusercontent.com/9455094/36543639-eb10f16c-17a9-11e8-99a0-3ae92609a150.png)

The selection arrow still does not scale, but that's not a big deal.

daschuer commented:
I think It does not work that good. Here are screen shots from Ubuntu Trusty:  

Master

100 %
![bildschirmfoto vom 2018-02-22 20 41 22](https://user-images.githubusercontent.com/1777442/36561157-35e55232-1813-11e8-8305-1b6462919402.png)

200 %
![bildschirmfoto vom 2018-02-22 20 43 35](https://user-images.githubusercontent.com/1777442/36561125-1e0abcb0-1813-11e8-8005-edac181ab614.png)

50 %
![bildschirmfoto vom 2018-02-22 20 42 37](https://user-images.githubusercontent.com/1777442/36561107-0fc35e96-1813-11e8-9605-094683fad658.png)

This PR

100 %
![bildschirmfoto vom 2018-02-22 20 35 50](https://user-images.githubusercontent.com/1777442/36561256-81988d02-1813-11e8-867f-03f660428020.png)

200 %
![bildschirmfoto vom 2018-02-22 20 36 40](https://user-images.githubusercontent.com/1777442/36561202-5d2c2d16-1813-11e8-8996-9ca71ea51388.png)

50 %
![bildschirmfoto vom 2018-02-22 20 34 52](https://user-images.githubusercontent.com/1777442/36561236-6e50a4d2-1813-11e8-87d4-c141ff7ba557.png)

Be-ing commented:
I think the issue you are seeing is not related to this PR. I could only reproduce your screenshots by loading the graphic EQ effect and shrinking the window to the minimum width. For those screenshots with master, did you test with the window wider than the minimum? This PR fixes the issue of effect names for most effects being elided when scaling above 100%.

daschuer commented:
I did not resize the Mixxx window.
Interestingly, there is some space left right to the EQ parameter knobs.
In the original version the box has always almost the same size, this is not intended. 
However in your version the box seems to totally forget it's size.

Be-ing commented:
>  Interestingly, there is some space left right to the EQ parameter knobs.

Hmm, that is interesting...

daschuer commented:
I have just tested it again, and it seem that the space left to the Mix knob does not scale well. 
However, I can not reproduce the issue 50 % issue shown above. It was probably caused be the selected effect on the right side.  

In Tango, the text escapes the combobox with 200%:
![bildschirmfoto vom 2018-02-23 20 44 31](https://user-images.githubusercontent.com/1777442/36613973-683b2b8a-18db-11e8-9b0f-073e9a8b0dc9.png)

I am not sure what is a skin issue and what is an issue of this PR. 

Normally the widgets are already sized in 
LegacySkinParser::setupSize()
I am afraid that your solution is just a hack around an other skin issue. 
What happen with the "-1" 
<MinimumSize>-1,65</MinimumSize>
Did you try to set explicit a width here? 

Do we have a general issue with eliding and widget scaling?

Be-ing commented:
> Normally the widgets are already sized in
LegacySkinParser::setupSize()

Ah, thanks for pointing that out. I had tried to find the code that handled that before but did not find it. I will try experimenting with setting the size to various values in the skin.

Be-ing commented:
This is quite bizarre. Somehow the setting of the widget's size in LegacySkinParser::setupSize is having no effect on WEffectSelector.

Be-ing committed:
fix constructors of WEffectSelector and WBeatSpinBox

The improper construction of the parent WBaseWidget class was causing
some strange issues. Sizes specified by the skin were ignored for
WEffectSelector and neither widget had the correct tooltip in developer
mode.

Be-ing committed:
Deere: explicitly set size for effect selectors

This is required for skin scaling to work well.

Be-ing committed:
LateNight: fix EffectSelector scaling

Be-ing commented:
I found the bug. WEffectSelector (as well as WBeatSpinBox) were being constructed incorrectly. A pointer to the parent widget was being passed to the WBaseWidget constructor instead of `this`. I am not entirely sure why, but somehow this caused the size changes applied by LegacySkinParser::setupSize to not affect WEffectSelector.

Now that the code for the widgets is fixed, the sizes set by the skin are scaled properly. Deere did not have the EffectSelector size set explicitly, so I had to set that. I had to make minor adjustments to the QSS in Tango and LateNight for the dropdown menu to scale properly. Here are some screenshots:

Deere:
![image](https://user-images.githubusercontent.com/9455094/36635750-538b2046-1980-11e8-9ee7-4527b2af5baa.png)

LateNight:
![image](https://user-images.githubusercontent.com/9455094/36636111-2e81f464-1985-11e8-8ecb-fe70a449595e.png)

Tango:
![image](https://user-images.githubusercontent.com/9455094/36636102-15ca1d70-1985-11e8-80fa-cf92d83a417c.png)

Be-ing commented on res/skins/LateNight/style.qss:
```diff
@@ -1462,10 +1458,8 @@ QSpinBox:editable {
-  min-width: 40px;
-  max-width: 40px;
+  height: 20px;
+  width: 40px;
```
I can revert d9e66ff to avoid conflicts with your branch.

ronso0 commented:
> I can revert d9e66ff to avoid conflicts with your branch

Yes, that would nice!
OT: I fail to find the messages I got notified of via email. Sometimes I them scattered across Conversation and Reviews, but lately they seem to be gone... Any idea?

Be-ing commented:
>  OT: I fail to find the messages I got notified of via email. Sometimes I them scattered across Conversation and Reviews, but lately they seem to be gone... Any idea?

GitHub doesn't send notifications when a pull request branch is rebased :/

ronso0 commented:
check #1527 with fixes for Tango

Be-ing committed:
Tango: fix EffectSelector scaling

Be-ing commented:
Rebased on 2.1 branch to incorporate fixes from #1527.

Be-ing commented:
@daschuer could you take another look at this?

daschuer commented:
LGTM Thank you.

Be-ing commented:
Thank you for review.
